### DISCUSSION TITLE
How to get full URL including parameters?

### DISCUSSION THREAD

**Discussion by yevon** [PostgREST/postgrest#2474] [Q&A] [answered]
Just a doubt about how can I get the full url address including parameters?

In the documentation I have seen that I can get current address without parameters like this:

`current_setting('request.path', true)`

But I didn't see any mention about how to get the full address including parameters.

**Comment by steve-chavez** [comment]
Yes, only the `path` is available, query parameters are not. There is a related PR https://github.com/PostgREST/postgrest/pull/1710

For now you can get the parameters with [database logging](https://postgrest.org/en/stable/admin.html#database-logs)

**Comment by yevon** [comment]
Any way I can do this with sql / pgsql?

**Reply by steve-chavez** [reply]
You mean access the db logs?

If that's the case, I think [csv logging](https://www.postgresql.org/docs/current/runtime-config-logging.html#RUNTIME-CONFIG-LOGGING-CSVLOG) plus [a foreign data wrapper](https://www.postgresql.org/docs/current/file-fdw.html#id-1.11.7.23.14) would work.

**Comment by yevon** [comment, marked as answer]
Finally solved this temporarily until I can access to the full url through params. I modified the nginx proxy to add the full url with parameters to a custom header, and then accessing to that through postgrest is easy:

nginx:
`proxy_set_header Custom-Header §uri?§args;`

pgsql:
`current_setting('request.headers', true)::json->>'custom-header)`
